File: android/app/src/main/java/com/smartexpense/android/presentation/util/UserManager.java
package com.smartexpense.android.presentation.util;

import android.content.Context;
import android.content.SharedPreferences;
import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.Locale;

public class UserManager {
    private static final String PREF_NAME = "set_user_pref";
    private static final String KEY_USER_NAME = "key_user_name";
    private static final String KEY_USER_EMAIL = "key_user_email";
    private static final String KEY_USER_PHONE = "key_user_phone";
    private static final String KEY_AVATAR_EMOJI = "key_avatar_emoji";
    private static final String KEY_MONTHLY_BUDGET = "key_monthly_budget";
    private static final String KEY_STREAK_DAYS = "key_streak_days";
    private static final String KEY_TOTAL_BILLS = "key_total_bills";

    private static final String DEFAULT_USER_NAME = "Nguyễn Văn A";
    private static final String DEFAULT_USER_EMAIL = "user@smartexpense.app";
    private static final String DEFAULT_USER_PHONE = "0912 345 678";
    private static final String DEFAULT_AVATAR_EMOJI = "⭐";
    private static final long DEFAULT_MONTHLY_BUDGET = 5000000L;

    public static String getUserName(Context context) {
        if (context == null) return DEFAULT_USER_NAME;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getString(KEY_USER_NAME, DEFAULT_USER_NAME);
    }

    public static void setUserName(Context context, String name) {
        if (context == null || name == null || name.trim().isEmpty()) return;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_USER_NAME, name.trim()).apply();
    }

    public static String getUserEmail(Context context) {
        if (context == null) return DEFAULT_USER_EMAIL;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getString(KEY_USER_EMAIL, DEFAULT_USER_EMAIL);
    }

    public static void setUserEmail(Context context, String email) {
        if (context == null || email == null || email.trim().isEmpty()) return;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_USER_EMAIL, email.trim()).apply();
    }

    public static String getUserPhone(Context context) {
        if (context == null) return DEFAULT_USER_PHONE;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getString(KEY_USER_PHONE, DEFAULT_USER_PHONE);
    }

    public static void setUserPhone(Context context, String phone) {
        if (context == null || phone == null || phone.trim().isEmpty()) return;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_USER_PHONE, phone.trim()).apply();
    }

    public static String getAvatarEmoji(Context context) {
        if (context == null) return DEFAULT_AVATAR_EMOJI;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getString(KEY_AVATAR_EMOJI, DEFAULT_AVATAR_EMOJI);
    }

    public static void setAvatarEmoji(Context context, String emoji) {
        if (context == null || emoji == null) return;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_AVATAR_EMOJI, emoji.trim()).apply();
    }

    public static long getMonthlyBudget(Context context) {
        if (context == null) return DEFAULT_MONTHLY_BUDGET;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getLong(KEY_MONTHLY_BUDGET, DEFAULT_MONTHLY_BUDGET);
    }

    public static void setMonthlyBudget(Context context, long budget) {
        if (context == null || budget <= 0) return;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putLong(KEY_MONTHLY_BUDGET, budget).apply();
    }

    public static String formatCurrency(long amount) {
        DecimalFormatSymbols symbols = new DecimalFormatSymbols(new Locale("vi", "VN"));
        symbols.setGroupingSeparator('.');
        DecimalFormat formatter = new DecimalFormat("#,###", symbols);
        return formatter.format(amount) + " ₫";
    }

    public static int getStreakDays(Context context) {
        if (context == null) return 5;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getInt(KEY_STREAK_DAYS, 5);
    }

    public static int getTotalBillsCount(Context context) {
        if (context == null) return 24;
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return prefs.getInt(KEY_TOTAL_BILLS, 24);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/presentation/util/UserManager.kt
package com.smartexpense.android.presentation.util

import android.content.Context
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.Locale

object UserManager {
    private const val PREF_NAME = "set_user_pref"
    private const val KEY_USER_NAME = "key_user_name"
    private const val KEY_USER_EMAIL = "key_user_email"
    private const val KEY_USER_PHONE = "key_user_phone"
    private const val KEY_AVATAR_EMOJI = "key_avatar_emoji"
    private const val KEY_MONTHLY_BUDGET = "key_monthly_budget"
    private const val KEY_STREAK_DAYS = "key_streak_days"
    private const val KEY_TOTAL_BILLS = "key_total_bills"
    private const val KEY_AUTH_TOKEN = "key_auth_token"
    private const val KEY_REFRESH_TOKEN = "key_auth_refresh_token"

    private const val DEFAULT_USER_NAME = "Nguyễn Văn A"
    private const val DEFAULT_USER_EMAIL = "user@smartexpense.app"
    private const val DEFAULT_USER_PHONE = "0912 345 678"
    private const val DEFAULT_AVATAR_EMOJI = "✨"
    private const val DEFAULT_MONTHLY_BUDGET = 5000000L

    fun saveToken(context: Context, token: String) {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_AUTH_TOKEN, token).apply()
    }

    fun saveRefreshToken(context: Context, token: String) {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_REFRESH_TOKEN, token).apply()
    }

    fun getToken(context: Context): String? {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_AUTH_TOKEN, null)
    }

    fun getRefreshToken(context: Context): String? {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_REFRESH_TOKEN, null)
    }

    fun clearToken(context: Context) {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit()
            .remove(KEY_AUTH_TOKEN)
            .remove(KEY_REFRESH_TOKEN)
            .apply()
    }

    fun getUserName(context: Context?): String {
        if (context == null) return DEFAULT_USER_NAME
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_USER_NAME, DEFAULT_USER_NAME) ?: DEFAULT_USER_NAME
    }

    fun setUserName(context: Context?, name: String?) {
        if (context == null || name.isNullOrBlank()) return
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_USER_NAME, name.trim()).apply()
    }

    fun getUserEmail(context: Context?): String {
        if (context == null) return DEFAULT_USER_EMAIL
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_USER_EMAIL, DEFAULT_USER_EMAIL) ?: DEFAULT_USER_EMAIL
    }

    fun setUserEmail(context: Context?, email: String?) {
        if (context == null || email.isNullOrBlank()) return
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_USER_EMAIL, email.trim()).apply()
    }

    fun getUserPhone(context: Context?): String {
        if (context == null) return DEFAULT_USER_PHONE
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_USER_PHONE, DEFAULT_USER_PHONE) ?: DEFAULT_USER_PHONE
    }

    fun setUserPhone(context: Context?, phone: String?) {
        if (context == null || phone.isNullOrBlank()) return
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_USER_PHONE, phone.trim()).apply()
    }

    fun getAvatarEmoji(context: Context?): String {
        if (context == null) return DEFAULT_AVATAR_EMOJI
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_AVATAR_EMOJI, DEFAULT_AVATAR_EMOJI) ?: DEFAULT_AVATAR_EMOJI
    }

    fun setAvatarEmoji(context: Context?, emoji: String?) {
        if (context == null || emoji == null) return
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_AVATAR_EMOJI, emoji.trim()).apply()
    }

    fun getMonthlyBudget(context: Context?): Long {
        if (context == null) return DEFAULT_MONTHLY_BUDGET
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getLong(KEY_MONTHLY_BUDGET, DEFAULT_MONTHLY_BUDGET)
    }

    fun setMonthlyBudget(context: Context?, budget: Long) {
        if (context == null || budget <= 0) return
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putLong(KEY_MONTHLY_BUDGET, budget).apply()
    }

    fun formatCurrency(amount: Long): String {
        val symbols = DecimalFormatSymbols(Locale("vi", "VN")).apply {
            groupingSeparator = '.'
        }
        val formatter = DecimalFormat("#,###", symbols)
        return "${formatter.format(amount)} ₫"
    }

    fun getStreakDays(context: Context?): Int {
        if (context == null) return 5
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getInt(KEY_STREAK_DAYS, 5)
    }

    fun getTotalBillsCount(context: Context?): Int {
        if (context == null) return 24
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        return prefs.getInt(KEY_TOTAL_BILLS, 24)
    }
}
